unityfiles/Assets/Scripts/Game/Controls/AimControl.cs

```diff
@@ -70,21 +69,83 @@ public class AimControl {
+
+        if (toSqr > 0){
+            float sgnX = (dy <0) ? -1:1;
+            float xcord1 = (D*dy + sgnX*dx*Mathf.Sqrt(toSqr))/Mathf.Pow(dr,2);
+            float xcord2 = (D*dy - sgnX*dx*Mathf.Sqrt(toSqr))/Mathf.Pow(dr,2);
+
+            float ycord1 = (-D*dx + Mathf.Abs(dy)*Mathf.Sqrt(toSqr))/Mathf.Pow(dr,2);
+            float ycord2 = (-D*dx - Mathf.Abs(dy)*Mathf.Sqrt(toSqr))/Mathf.Pow(dr,2);
+            
+            Vector2 v_mp_pos1 = new Vector2(V2.x, V2.y) - new Vector2(xcord1, ycord1);
+            Vector2 v_mp_pos2 = new Vector2(V2.x, V2.y) - new Vector2(xcord2, ycord2);
+
+            if (v_mp_pos1.magnitude > v_mp_pos2.magnitude){
+                xcord = xcord2;
+                ycord = ycord2;
+            } else {
+                xcord = xcord1;
+                ycord = ycord1;
+            }
+
+            float ang2 = this.RadianToDegree(Math.Atan2(ycord, xcord));
+            float deltaAng =ang2 - z_angle;
+            z_angle -= deltaAng;
+        } else {
+            float ang2 = this.RadianToDegree(Math.Atan2(V2.y, V2.x));
+            float deltaAng =ang2 - z_angle;
+            z_angle -= deltaAng;
+        }
 
         if (debug) {
-            Debug.DrawLine(Vector3.zero, norm_GP, Color.red);
-            Debug.DrawLine(Vector3.zero, v_GP_MP, Color.green);
-            Debug.DrawLine(Vector3.zero, Vector3.Project(v_GP_MP, norm_GP), Color.yellow);
-            Debug.DrawLine(Vector3.zero, norm_GP_Prj_GPMP - v_GP_MP, Color.blue);
+            Debug.Log($"Z ANG     {z_angle}");
+            Debug.Log($"XCORD   {xcord}  YCORD   {ycord}");
+            Debug.Log($"the math   {Mathf.Pow(r,2)*Mathf.Pow(dr,2)-Mathf.Pow(D,2)}");
+            //Debug.Log($"MOUSE  X    {mousePoint.x - helperRotPoint.transform.position.x}  Y   {mousePoint.y -helperRotPoint.transform.position.y}");
+
+            Debug.DrawLine(helperRotPoint.transform.position,helperRotPoint.transform.position+ V2, Color.red);
+            Debug.DrawLine(Vector3.zero, V2, Color.red);
+
+            Debug.DrawLine(helperRotPoint.transform.position,helperRotPoint.transform.position+ v_RP_GP, Color.green);
+            //Debug.DrawLine(helperRotPoint.transform.position,helperRotPoint.transform.position+ new Vector3(xcord1,ycord,0), Color.yellow);
+            Debug.DrawLine(helperRotPoint.transform.position,helperRotPoint.transform.position+v_RP_GP_Prj_GPMP, Color.blue);
+            // Debug.DrawLine(mousePoint,mousePoint + (norm_GP_Prj_GPMP - v_GP_MP), Color.blue);
             Debug.DrawLine(RotationPoint.position, Camera.main.ScreenToWorldPoint(Input.mousePosition), Color.magenta);
+
+            Debug.DrawRay(RotationPoint.transform.position, RotationPoint.transform.right * 100, Color.gray);
             Debug.DrawRay(FirePoint.transform.position, FirePoint.transform.right * 100, Color.cyan);
+            //Debug.DrawRay(FirePoint.transform.position, helperAimPoint.transform.right * 100, Color.white);
```

Comment:
slett ubrukt kode

Reply:
Å fjerne denne blokken vil jøre debuging veldig vanskelig frammover. siden det er fungsjonallitet for og skru debugging av og på tenker jeg at denne får stå